src/wp-includes/rest-api/endpoints/class-wp-rest-application-passwords-controller.php

```diff
@@ -0,0 +1,656 @@
+	 *
+	 * @since ?.?.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to delete the item, WP_Error object otherwise.
+	 */
+	public function delete_items_permissions_check( $request ) {
+		return $this->do_permissions_check( $request );
+	}
+
+	/**
+	 * Deletes all application passwords.
+	 *
+	 * @since ?.?.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_items( $request ) {
+		$user = $this->get_user( $request );
+
+		if ( is_wp_error( $user ) ) {
+			return $user;
+		}
+
+		$deleted = WP_Application_Passwords::delete_all_application_passwords( $user->ID );
+
+		if ( is_wp_error( $deleted ) ) {
+			return $deleted;
+		}
+
+		return new WP_REST_Response(
+			array(
+				'deleted' => true,
+				'count'   => $deleted,
+			)
+		);
+	}
+
+	/**
+	 * Checks if a given request has access to delete a specific application password.
+	 *
+	 * @since ?.?.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to delete the item, WP_Error object otherwise.
+	 */
+	public function delete_item_permissions_check( $request ) {
+		return $this->do_permissions_check( $request );
+	}
+
+	/**
+	 * Deletes one application password.
+	 *
+	 * @since ?.?.0
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function delete_item( $request ) {
+		$user = $this->get_user( $request );
+
+		if ( is_wp_error( $user ) ) {
+			return $user;
+		}
+
+		$password = $this->get_application_password( $request );
+
+		if ( is_wp_error( $password ) ) {
+			return $password;
+		}
+
+		$request->set_param( 'context', 'edit' );
+		$previous = $this->prepare_item_for_response( $password, $request );
+		$deleted  = WP_Application_Passwords::delete_application_password( $user->ID, $password['uuid'] );
+
+		if ( is_wp_error( $deleted ) ) {
+			return $deleted;
+		}
+
+		return new WP_REST_Response(
+			array(
+				'deleted'  => true,
+				'previous' => $previous->get_data(),
+			)
+		);
+	}
+
+	/**
+	 * Performs a permissions check for the request.
+	 *
+	 * @since ?.?.0
+	 *
+	 * @param WP_REST_Request $request
+	 * @return true|WP_Error
+	 */
+	protected function do_permissions_check( $request ) {
+		$user = $this->get_user( $request );
+
+		if ( is_wp_error( $user ) ) {
+			return $user;
+		}
+
+		if ( ! current_user_can( 'edit_user', $user->ID ) ) {
+			return new WP_Error(
+				'rest_cannot_manage_application_passwords',
+				__( 'Sorry, you are not allowed to manage application passwords for this user.' ),
+				array( 'status' => rest_authorization_required_code() )
+			);
+		}
```

Comment:
Is this a satisfactory permission check? Does it make sense that `edit_user` implies `manage application passwords`, or should that be a separate permission bit?